modules/hal_nordic/nrfx/nrfx_config.h

```diff
@@ -197,57 +198,80 @@
 #define NRFX_SAADC_ENABLED 1
 #endif
 
+#define SPI0_NODE DT_NODELABEL(spi0)
+#define SPI1_NODE DT_NODELABEL(spi1)
+#define SPI2_NODE DT_NODELABEL(spi2)
+#define SPI3_NODE DT_NODELABEL(spi3)
+#define SPI4_NODE DT_NODELABEL(spi4)
+
 #ifdef CONFIG_NRFX_SPI
 #define NRFX_SPI_ENABLED 1
-#endif
-#ifdef CONFIG_NRFX_SPI0
+
+#if DT_NODE_HAS_COMPAT_STATUS(SPI0, nordic_nrf_spi, okay)
```

Comment:
This cannot be done this way. We need to keep the possibility of using nrfx drivers directly for some instances (the corresponding nodes in devicetree are disabled then) and Zephyr shims for others.
In my opinion we should keep options `NRFX_SPIx`, `NRFX_SPIMx`, and `NRFX_SPISx`, and select them in `drivers/spi/Kconfig.nrfx` when a given instance is enabled in devicetree with the corresponding compatible property.